Celeste.Mod.mm/Mod/Everest/Emoji.cs

```diff
@@ -146,10 +148,58 @@ public static bool IsMonochrome(char c)
                 return text;
             // TODO: This trashes the GC and doesn't allow escaping!
             lock (_IDs) {
-                foreach (KeyValuePair<string, int> kvp in _IDs)
-                    text = text.Replace(":" + kvp.Key + ":", ((char) (Start + kvp.Value)).ToString());
+                StringBuilder resultBuilder = new StringBuilder();
+                int appendStartIndex = 0;
+                int head = -1, tail = 0;
+                // suppose text is "aaaa:1111:2222:bbbb", and only ":2222:" is emoji name
+                // H = head, T = tail, S = appendStartIndex
+                while (tail < text.Length) {
+                    if (text[tail] == ':') {
+                        if (head >= 0 && text[head] == ':') {
+                            /*
+                                    aaaa:1111:2222:bbbb
+                                (2) ^S  ^H   ^T   ^
+                                (4) ^S       ^H   ^T
+                                now head and tail are pointing to colons so we need to check if the text inside colons is an emoji name
+                            */
+                            string name = text.Substring(head + 1, (tail - 1) - (head + 1) + 1);
+                            bool exist = _IDs.TryGetValue(name, out int value);
+                            if (exist) {
```

Comment:
These lines can be merged without sacrificing readability
```suggestion
                            if (_IDs.TryGetValue(name, out int value)) {
```